src/core/algorithms/dc/parser/dc_parser.cpp

```diff
@@ -0,0 +1,168 @@
+#include "algorithms/dc/parser/dc_parser.h"
+
+#include <stdexcept>
+
+#include <boost/algorithm/string.hpp>
+
+#include "algorithms/dc/model/dc.h"
+#include "algorithms/dc/model/operator.h"
+#include "algorithms/dc/model/predicate.h"
+#include "model/table/column_layout_relation_data.h"
+#include "model/table/typed_column_data.h"
+
+namespace mo = model;
+
+namespace {
+
+using namespace algos::dc;
+
+std::unique_ptr<ColumnOperand> GetOperand(ColumnOperand const& operand, std::string str_val) {
+    mo::Type const* type = operand.GetType();
+    return std::make_unique<ColumnOperand>(str_val, type);
+}
+
+}  // namespace
+
+namespace algos::dc {
+
+DCParser::DCParser(std::string dc_string, ColumnLayoutRelationData const* relation,
+                   std::vector<model::TypedColumnData> const& data)
+    : relation_(relation),
+      data_(data),
+      dc_string_(std::move(dc_string)),
+      has_next_predicate_(true),
+      cur_(0) {
+    str_operators_.reserve(str_operators_.size());
+    for (std::pair<frozen::string const, algos::dc::OperatorType> const& op_it :
+         Operator::kStringToOperatorType) {
+        std::string str_op = op_it.first.data();
```

Comment:
maybe smth like
```suggestion
    for (auto const& [str, _] : Operator::kStringToOperatorType) {
        std::string str_op = str.data();
```
but with name better than just `str`